stacks-signer/src/config.rs

```diff
@@ -38,25 +43,70 @@ pub enum ConfigError {
     BadField(String, String),
 }
 
+#[derive(serde::Deserialize, Debug, Clone)]
+#[serde(rename_all = "lowercase")]
+/// The Stacks network to use.
+pub enum Network {
+    /// The mainnet network
+    Mainnet,
+    /// The testnet network
+    Testnet,
+}
+
 /// The parsed configuration for the signer
 pub struct Config {
     /// endpoint to the stacks node
     pub node_host: SocketAddr,
+    /// endpoint to the stackerdb receiver
+    pub endpoint: SocketAddr,
     /// smart contract that controls the target stackerdb
     pub stackerdb_contract_id: QualifiedContractIdentifier,
-    /// the private key used to sign blocks, chunks, and transactions
-    pub private_key: StacksPrivateKey,
+    /// The Scalar representation of the private key for signer communication
+    pub message_private_key: Scalar,
+    /// The signer's Stacks private key
+    pub stacks_private_key: StacksPrivateKey,
+    /// The network to use. One of "mainnet" or "testnet".
+    pub network: Network,
+    /// The signer ID and key ids mapped to a pulbic key
+    pub signer_ids_public_keys: PublicKeys,
+    /// The signer IDs mapped to their Key IDs
+    pub signer_key_ids: SignerKeyIds,
+    /// This signer's ID
+    pub signer_id: u32,
+    /// The time to wait for a response from the stacker-db instance
+    pub event_timeout: Duration,
+}
+
+/// Internal struct for loading up the config file signer data
+#[derive(Clone, Deserialize, Default, Debug)]
+struct RawSigners {
+    pub public_key: String,
+    pub key_ids: Vec<u32>,
 }
 
 /// Internal struct for loading up the config file
 #[derive(Deserialize, Debug)]
 struct RawConfigFile {
     /// endpoint to stacks node
     pub node_host: String,
+    /// endpoint to stackerdb receiver
+    pub endpoint: String,
     /// contract identifier
     pub stackerdb_contract_id: String,
-    /// the private key used to sign blocks, chunks, and transactions in hexademical format
-    pub private_key: String,
+    /// the 32 byte ECDSA private key used to sign blocks, chunks, and transactions
+    pub message_private_key: String,
+    /// The hex representation of the signer's Stacks private key
+    pub stacks_private_key: String,
+    /// The network to use. One of "mainnet" or "testnet".
+    pub network: Network,
+    // TODO: Optionally retrieve the signers from the pox contract
```

Comment:
Doesn't have to be done in this PR ofc, but it's not optional ;)  The default behavior will be to obtain the signers' public keys from `pox-4` (or, at the user's discretion, a conforming smart contract that could be used to mock `pox-4` for testing).

Reply:
Just commenting here so it gets on y'all's radar.

Reply:
Only reason I have this marked optional is because I could see for testing purposes that we should be able to provide this information in a config rather than just always from the pox contract.